src/rgw/rgw_rest_s3.h

```diff
@@ -119,11 +119,10 @@ class RGWListBucket_ObjStore_S3 : public RGWListBucket_ObjStore {
   void send_versioned_response();
 };
 
-class RGWListBucket_ObjStore_S3v2 : public RGWListBucket_ObjStore {
-  bool objs_container, fetchOwner;
+class RGWListBucket_ObjStore_S3v2 : public RGWListBucket_ObjStore_S3 {
+  bool fetchOwner;
 public:
-  RGWListBucket_ObjStore_S3v2() : objs_container(false), fetchOwner(false) {
-    default_max = 1000;
+  RGWListBucket_ObjStore_S3v2() :  fetchOwner(false) {
   }
   ~RGWListBucket_ObjStore_S3v2() override {}
 
```

Comment:
You can invoke RGWListBucket_ObjStore_S3::get_params from RGWListBucket_ObjStore_S3v2::get_params(), and add things specific to RGWListBucket_ObjStore_S3v2 after that. Same for other methods like send_response() and send_versioned_response()

Reply:
Thanks @pritha-srivastava

Reply:
Have missed refactoring send_versioned_response()? Also, can you indent your code properly?

Reply:
Added